src/components/DatePicker/__tests__/DatePicker.spec.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+import MockDate from 'mockdate';
+import { DatePicker } from '../DatePicker';
+
+describe('DatePicker', () => {
+    let view;
+    const handleChange = jest.fn();
+
+    const minDate = new Date('2021-11-23');
+    const maxDate = new Date('2023-03-04');
+
+    const openCalendar = async (user) => {
+        await user.click(screen.getByRole('textbox'));
+    };
+
+    beforeAll(() => {
+        MockDate.set('2021-12-24T13:00:00.000Z');
+    });
+
+    afterAll(() => {
+        MockDate.reset();
+    });
+
+    beforeEach(() => {
+        view = render(<DatePicker onChange={handleChange} minDate={minDate} maxDate={maxDate} />);
+    });
+
+    it('should render correctly', () => {
+        expect(view.container).toMatchSnapshot();
+        expect(screen.getByRole('textbox')).toBeInTheDocument();
+    });
+
+    it('should render calendar header correctly when calendar is open', async () => {
+        const user = userEvent.setup();
+        await openCalendar(user);
+        // taking full snapshot of the calendar fails in CD/CI.
+        // Some small time difference in dates - needs further investigation.
+        // Since testing 3rd party library is not a good practice anyway,
+        // we will just check that our custom header is correct
+        // screen.getByRole('option');
+        expect(screen.getByRole('presentation')).toBeInTheDocument();
+    });
+
+    it('should set min and max date on calendar even if they are no originally passed as props', async () => {
+        const user = userEvent.setup();
+        // this is to make sure the calendar navigation and the year selection in the header are always aligned
+        await openCalendar(user);
+
+        expect(user);
```

Comment:
This will be always green or am I mistaken?

Reply:
Yes, but it's not clear what we expect here. I think we should add
expect(screen.getByText(minDate.getFullYear())).toBeInTheDocument();
expect(screen.getByText(maxDate.getFullYear())).toBeInTheDocument();

Reply:
I've update it, could you please check it

Reply:
But do we need this `expect(user)` expectation? What is the reasoning for that?

Reply:
:+1: